### DISCUSSION TITLE
Option to restart a container before pausing.

### DISCUSSION THREAD

**Discussion by Nerokor** [Fuglesteg/Timid#3] [Ideas]
I'm proposing an option to restart a container, then wait a certain length of time before pausing. One use case for wanting this is a Palworld server, which is notorious for having a bad memory leak and needing to be constantly restarted. Having the container restart after no network activity is detected, waiting x amount of time for it to boot up, and then pausing would keep it ready to go and make sure memory usages doesn't get out of control.

**Comment by Fuglesteg** [comment]
That's an interesting idea. I would be willing to implement another environment variable, something along the lines of `TIMID_RESTART_BEFORE_PAUSE_DELAY`,
to specify the amount of boot time given to the container when restarting before pausing. The use case does seem pretty niche though so I would be interested in exploring some more general way of implementing this. I am currently working on an HTTP API that would expose actions like restarting and pausing containers. Perhaps a more general solution would be to expose some kind of easy to define behaviors using the API. Any thoughts?
